### PR TITLE
Provena Minor Release v2.1

### PR DESCRIPTION
**Pull request by PeterBaker0**
# Provena PR (Minor): v2.1 Minor Release

## Major Features
- Dataset metadata extension 
  - includes a suite of new optional fields to enrich dataset metadata, reaching closer to ISO 19115 compatibility
  - also greatly improves how JSON schema is generated form the dataset metadata Pydantic model by returning to a single source of truth using Pydantic annotations
  - adds overrides for optional fields in the RJSF form to more clearly annotate optional inclusions and avoid issues with inclusion of empty objects `{}` which are not needed in the registered payload

## Minor Features
- Right clickable cards in the Registry UI list view
- New interface for directly entering the ID of an item in the registry rather than searching
- Better handling of long descriptions for datasets
- Improved styling for new activity subtypes Create and Version

## Bug Fixes
- Use of utf-16 encoding for email text in email service, enabling emojis etc
- Improved auth error handling in user link workflow

## Other
- Improved testing for dataset releases
- Update links in docs

# Migrations (v2.0 -> v2.1)

## Additional integration test user

In the case where your deployment uses integration tests, you will need to create an additional test user with the appropriate permissions, and update the existing integration tests secret to include this user/pass.

### Steps

1. Creating the new user in the Keycloak Console.

Log into the Keycloak console and navigate to "users" on the left tab under heading 'manage'.

On the right, click 'add user'.

Specify the username of the new test bot, its a good idea to include the permissions it will receive. For example, "integration-test-bot-read-write-3". Click save.

Got back to the Users tab and search for the bot you just created, click on it, then navigate to its credentials tab and set its password to something and save this password for step 3. We recommend using a strong randomly generated password. IMPORTANT: Turn off the "temporary" setting as this is a test bot and not a real user. Click Reset password.

Now we have a new bot user, we need to give it the permissions we want.

Click on Groups tab on the Top panel next to the "Credentials" tab we have already clicked on when we just reset the password.

Assign the bot the level of system permission desired. For a bot that is performing read/write actions in the integration test, we join Entity-registry group "Write". To do this, select the group from the "available groups" panel, then 'Join' once the desired group is selected.

Now the bot user is set up with a username, password and desired permissions, we need to now use its credentials in the integration tests.

2. Using credentials in the Pipeline Integration tests

Credentials are automatically injected at pipeline runtime to the environment that runs the tests. They are injected from AWS SecretManager -  we need to add the new username and password for our new bot to the secret that stores the integration test bot credentials.

Open up the Secrets Manager in the AWS console. Search for "integration" and you should see a secret called "<stage>_integration_test_bots". (e.g. dev_integration_test_bots). Click on it, then on 'retrieve secret value', then you can click on 'edit'.

Add new keys and values for the username and password of the new bot. IMPORTANT: Ensure that they are named exactly "username3" and "password3", as this is the key expected by the CDK infrastructure code when the integration test environment variables are sourced from the secret at run time.

3. (**optional**) Using credentials in the locally run integration tests

In the source code of your repo, navigate to the env file used for integration testing: "provena/tests/integration/.env". 

Look at the template-env file in the same directory which has new fields for:

SYSTEM_WRITE_USERNAME_3=""
SYSTEM_WRITE_PASSWORD_3=""

Add these fields with the actual values in your .env file. The tests will read these in along with other environment variables at run time. The username and password is what we entered when creating the new bot in step 1.

## Contributors

- @Rosspet 
- @xinyuhoulucy 
- @PeterBaker0 

## Includes the following PRs:

RRAPIS-1511 (Minor): Right Clickable Items in Registry (#48)
RRAPIS-1492 (Bug Fix): Emoji email (#47)
RRAPIS-1524 (Major): Dataset Metadata Extension (#46)
RRAPIS-1493 (Minor): Dataset Release Tests (#44)
RRAPIS-1410 (Minor): Fetch item feature in registry UI (#43)
RRAPIS-1506 (Minor): Wrap dataset description with expand in dataset list and overview tab (#42)
RRAPIS-1498 (Minor): Create and version styling (#40)
RRAPIS-1374 (Minor): New-user-profile-identity-link-read-permission-problem (#39)
RRAPIS-1491 (Minor): Update-doc-links-in-all-uis (#38)

### CODE REVIEW

**Review by xinyuhoulucy — CHANGES_REQUESTED** [review]
Looks all good to me, just need to make sure the interface names in ```data-store-ui/src/shared-interfaces/registryModels.ts``` is correct as we expected.

File: data-store-ui/src/stores/registerOrModifyStore.ts [hunk 1]
```diff
@@ -1,22 +1,22 @@
...
+import lodashGet from "lodash.get";
+import lodashSet from "lodash.set";
 import { makeAutoObservable } from "mobx";
+import { LabelPathAndValue } from "pages/UpdateMetadata";
 import {
     DATA_STORE_API_ENDPOINTS,
     displayValidationError,
     requests,
 } from "react-libs";
+import { CollectionFormat } from "shared-interfaces/registryModels";
```
[the remark this answers is not in the document]
Comment (PeterBaker0) [reply data-store-ui/src/stores/registerOrModifyStore.ts:11]: Yes? I think it's fine - not sure what you mean. These import changes is just due to organise imports renaming/moving things around.

File: prov-ui/src/shared-interfaces/RegistryModels.ts [hunk 2]
```diff
@@ -90,59 +102,171 @@ export interface AuthTableEntry {
...
 }
 export interface AutomationSchedule {}
-export interface CollectionFormat {
-  associations: CollectionFormatAssociations;
-  dataset_info: CollectionFormatDatasetInfo;
-  approvals: CollectionFormatApprovals;
+/**
+ * This form collects metadata about the dataset you are registering as well as the registration itself. Please ensure the fields below are completed accurately.
+ */
+export interface DatasetMetadata {
```
[the remark this answers is not in the document]
Comment (PeterBaker0) [reply prov-ui/src/shared-interfaces/RegistryModels.ts:108]: Okay

**Review by xinyuhoulucy — APPROVED** [review]

File: data-store-ui/src/stores/registerOrModifyStore.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (PeterBaker0) [reply data-store-ui/src/stores/registerOrModifyStore.ts]: After discussion with you I understand the issue here - great pickup thanks.

Have fixed in subsequent update.

**Pull request merged**